platforms/access-bridge/src/services/stripe-service.ts

```diff
@@ -21,12 +21,28 @@ export class StripeService {
     });
   }
 
+  /**
+   * Retrieves a Stripe customer ID based on the email address.
+   * @param email The email address of the customer.
+   * @returns A Promise resolving to the customer ID or null if no customer is found.
+   * @throws Error if there is an issue searching stripe customers by email.
+   */
+  async getCustomerIdByEmail({ email }: { email: string }): Promise<string | null> {
+    // Search for customers by email using Stripe's API.
+    const customers = await this.stripe.customers.search({
+      query: `email:'${email}'`,
+    });
+
+    // Return the first customer's ID if available, otherwise null.
+    return customers.data.length ? customers.data[0].id : null;
```

Comment:
What does it mean if multiple are returned?

Reply:
I've updated the logic a little. I don;'t think it's possible to have more than 1 customers if we search by email, as we do. This is Stripe's search API and probably works with few characters only, instead of entire email so I assume that's why they have it structured like that.